crates/walrus-sdk/src/client/client_types.rs

```diff
@@ -694,6 +697,34 @@ impl WalrusStoreBlobMaybeFinished<BlobWithStatus> {
         drop(_guard);
         Ok(self)
     }
+
+    /// Returns the metadata and sliver pairs needed for an optimistic upload if eligible.
+    pub fn pending_upload_payload(
+        &self,
+        pending_uploads_enabled: bool,
+        store_args: &StoreArgs,
+        optimistic_upload_max_blob_bytes: u64,
+    ) -> Option<(VerifiedBlobMetadataWithId, Arc<Vec<SliverPair>>)> {
+        let WalrusStoreBlobState::Unfinished(blob_with_status) = &self.state else {
+            return None;
+        };
+        if blob_with_status.status.is_registered()
+            || !pending_uploads_enabled
+            || !store_args.store_optimizations.optimistic_uploads_enabled()
```

Comment:
Nit: pending_upload and optimistic_upload mean the same thing, is that correct? If so, can we converge them into one name?

Reply:
done